src/utilities/api-requests.ts

```diff
@@ -10,13 +10,26 @@ import axios from "axios";
  */
 export async function request<T>(uri: string, defaultValue: T): Promise<T> {
     let data = defaultValue;
-    await axios
-        .get(uri)
-        .then((res) => {
-            data = res.data as T;
-        })
-        .catch((error) => {
-            console.error(`Request to ${uri} failed with\n${error}`);
-        });
+    let attempt = 0;
+
+    while (data === defaultValue && attempt < 5) {
+        let response = await axios
```

Comment:
Codacy found an issue: [Identifier 'response' is never reassigned; use 'const' instead of 'let'.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=5953345)